File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) 
    {
        int size=nums1.length+nums2.length;
        int []nums3=new int[size];
        for(int i=0;i<nums1.length;i++)
            nums3[i]=nums1[i];
        for(int i=0;i<nums2.length;i++)
            nums3[i+nums1.length]=nums2[i];
        Arrays.sort(nums3);
        if(nums3.length%2!=0)
        {
            int c=nums3.length/2;
            double b=Math.ceil(c);
            int b2=(int)b;
            return nums3[b2];
        }
        else
        {
            int a=(nums3.length/2)-1;
            double res=(nums3[a]+nums3[a+1]);
            return res/2;
        }
        
    }
}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        
        n1 = len(nums1)
        n2 = len(nums2)
        
        if n2 < n1:
            return self.findMedianSortedArrays(nums2, nums1)
        
        leftPortionSize = (n1 + n2 + 1) >> 1
        
        low = 0
        high = n1
        
        while low <= high:
            
            mid = (low + high) >> 1
            
            mid1 = mid
            mid2 = leftPortionSize - mid1
            
            l1, l2 = float('-inf'), float('-inf')
            r1, r2 = float('inf'), float('inf')
            
            if mid1 < n1:
                r1 = nums1[mid1]
            
            if mid2 < n2:
                r2 = nums2[mid2]
            
            if mid1 - 1 >= 0:
                l1 = nums1[mid1 - 1]
            
            if mid2 - 1 >= 0:
                l2 = nums2[mid2 - 1]
            
            leftMax = max(l1, l2)
            rightMin = min(r1, r2)
            
            if leftMax <= rightMin:
                if (n1 + n2) % 2:
                    res = max(l1, l2) 
                else:
                    res = (max(l1, l2) + min(r1, r2)) / 2
                
            if l1 > r2:
                high = mid - 1
            
            else:
                low = mid + 1
                
        return res
            